lib/vre/service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module VRE
+  class Service
+    def get_token
+      conn = Faraday.new(
+        "#{Settings.veteran_readiness_and_education.auth_endpoint}?grant_type=client_credentials",
+        headers: { 'Authorization' => Settings.veteran_readiness_and_education.credentials }
+      )
+
+      request = conn.post
+
+      JSON.parse(request.body)['access_token']
+    end
+
+    def send_to_vre(payload, exception)
+      conn = Faraday.new(url: Settings.veteran_readiness_and_education.base_url)
+
+      response = conn.post do |req|
+        req.url Settings.veteran_readiness_and_education.ch_31_endpoint
+        req.headers['Authorization'] = "Bearer #{get_token}"
+        req.headers['Content-Type'] = 'application/json'
+        req.body = payload
+      end
+
+      response_body = JSON.parse(response.body)
+      return true if response_body['ErrorOccurred'] == false
+
+      raise exception
+    rescue exception => e
+      log_exception_to_sentry(
+        e,
+        {
+          intake_id: response_body['ApplicationIntake'],
+          error_message: response_body['ErrorMessage']
+        },
+        { team: 'vfs-ebenefits' }
+      )
+    end
```

Comment:
Typically we will extract configuration for a client to it's own class, i.e. `VRE::Configuration` and that would live at `lib/vre/configuration.rb`

In this file you can do Faraday configuration, like this:

```rb

# frozen_string_literal: true

require 'common/client/configuration/rest'

module VRE
  class Configuration < Common::Client::Configuration::REST
    def connection
      @conn ||= Faraday.new(base_path, headers: request_headers request: request_options) do |faraday|
        faraday.use :breakers
        faraday.use Faraday::Response::RaiseError
        faraday.response :betamocks if mock_enabled?
        faraday.response :snakecase, symbolize: false
        faraday.response :json, content_type: /\bjson/
        faraday.adapter Faraday.default_adapter
    end

    def request_headers
      { 
        'Authorization': "Bearer #{get_token}"
      }.freeze
   end

    def mock_enabled?
      Settings.veteran_readiness_and_education.mock_ch_31 || false
    end
    
    def base_path
      Settings.veteran_readiness_and_education.base_url
    end

    def service_name
      'VeteranReadinessEducation'
    end
  end
end
```

Then you can include this configuration file into your base service class, with a `require 'vre/configuration'` and using it by declaring `configuration VeteranReadinessEducation::Configuration` and call it like so:


```rb

# ...
def send_to_vre(payload, exception)
  perform(method, path, params, headers, options) # see lib/common/client/base.rb#L94
end
```